lib/services/wallet_connect_service.dart

```diff
@@ -355,6 +361,46 @@ class WalletConnectService {
           }
         });
 
+    web3Wallet!.registerRequestHandler(
+        chainId: Config.walletConnectChainId,
+        method: WcMethods.wSendTransaction,
+        handler: (topic, args) async {
+          final sessionRequest = web3Wallet!.pendingRequests
+              .getAll()
+              .where((e) =>
+                  e.method == WcMethods.wSendTransaction && e.topic == topic)
+              .last;
+
+          late RequestSendTransactionEvent event;
+
+          if (sendTransactionHandler == null) {
+            throw "sendQubicHandler is not set";
+          }
+          try {
+            event = RequestSendTransactionEvent.fromMap(
+                args, topic, sessionRequest.id);
+            event.validateOrThrow();
+
+            if (web3Wallet!.getActiveSessions().containsKey(topic)) {
```

Comment:
@ahmed-tarek-salem  I see these lines to find a session is executed in inside method execution. Can we create a method that all reuse?

Reply:
Fixed with 0e6d461 and e5994fa